src/connect/wrapMapToProps.ts

```diff
@@ -1,7 +1,16 @@
+import { Dispatch } from 'redux'
+
 import verifyPlainObject from '../utils/verifyPlainObject'
 
-export function wrapMapToPropsConstant(getConstant) {
-  return function initConstantSelector(dispatch, options) {
+export type MapToProps = {
+  (stateOrDispatch: any, ownProps?: any): any
+  dependsOnOwnProps: boolean
+}
+
+export function wrapMapToPropsConstant(
+  getConstant: (dispatch: Dispatch, options: any) => void
```

Comment:
:hand: Hmm. Despite the variable name, I don't _think_ what's being passed in here is actually `dispatch` all the time.  This gets called twice in the `mapDispatch` file and once in `mapState`:

`mapState`: 

```js
export function whenMapStateToPropsIsMissing(mapStateToProps) {
  return !mapStateToProps ? wrapMapToPropsConstant(() => ({})) : undefined
}
```

`mapDispatch`: 

```js
export function whenMapDispatchToPropsIsMissing(mapDispatchToProps) {
  return !mapDispatchToProps
    ? wrapMapToPropsConstant((dispatch) => ({ dispatch }))
    : undefined
}

export function whenMapDispatchToPropsIsObject(mapDispatchToProps) {
  return mapDispatchToProps && typeof mapDispatchToProps === 'object'
    ? wrapMapToPropsConstant((dispatch) =>
        bindActionCreators(mapDispatchToProps, dispatch)
      )
    : undefined
}
```

Based on that, it looks like what gets passed in is actually `state` when we're dealing with `mapState`, and `dispatch` when we're dealing with `mapDispatch`.  Also, there _aren't_ any `options` being passed in or used anywhere.

On the other hand, for the `mapState` case, we ignore what's being passed as the first arg anyway.

We can maybe leave `dispatch: Dispatch` for now, but I think we can remove `options` entirely.

Reply:
Thanks a lot for the feedback @markerikson 🙌 

So, I finally use `unknown` and `FixTypeLater`, let me know if it make sense for you. 

I noticed that other functions have a `stateOrDispatch` argument. So it make a lot of sense.

Reply:
When I step back a bit I found that there are too much `FixTypeLater`, WDYT ? I can dig a bit more (using break points) to better understand what's happening there.

Reply:
If you'd like to spend some time further analyzing exactly what's being passed through, sure, go ahead :)